scripts/train_tokenizer.py

```diff
@@ -0,0 +1,131 @@
+import argparse
+import os
+
+from datasets import load_dataset
+from tqdm.auto import tqdm
+from transformers import LlamaTokenizerFast
+from tokenizers import SentencePieceBPETokenizer
+from tqdm import trange
+
+from delphi.train.tokenizer import train_vocab, get_tokenizer_model_path
+
+
+def main(
+    vocab_size: int,
+    dataset_name: str,
+    train_size: float,
+    username: str,
+    token: str,
+    funct_test: bool = False,
+):
+    """
+    Trains a SentencePiece tokenizer on a dataset.
+    And uploads the resulting tokenizer to huggingface.
+    Args:
+    - vocab_size: The vocabulary size of the resulting tokenizer
+    - dataset_name: The name of the dataset from which validation set will be loaded
+    - train_size: The amount of the dataset that should be used for training
+    - username: Hugging Face API username
+    - token: Hugging Face API token
+    """
+    train_ds = load_dataset(dataset_name)["train"]
+    if train_size < 1.0:
+        train_ds = train_ds.train_test_split(train_size=train_size)["train"]
+        
+    tokenizer_model_path = get_tokenizer_model_path(
+        vocab_size=vocab_size,
+    )
```

Comment:
I don't think this should be a separate function

Reply:
removed